src/run/handlers/server.ts

```diff
@@ -128,19 +124,19 @@ export default async (request: Request, context: FutureContext) => {
       return new Response(body || null, response)
     }
 
-    if (context.waitUntil) {
-      context.waitUntil(requestContext.backgroundWorkPromise)
-    }
-
     const keepOpenUntilNextFullyRendered = new TransformStream({
       async flush() {
         // it's important to keep the stream open until the next handler has finished
         await nextHandlerPromise
-        if (!context.waitUntil) {
-          // if waitUntil is not available, we have to keep response stream open until background promises are resolved
-          // to ensure that all background work executes
-          await requestContext.backgroundWorkPromise
-        }
+
+        // Next.js relies on `close` event emitted by response to trigger running callback variant of `next/after`
+        // however @fastly/http-compute-js never actually emits that event - so we have to emit it ourselves,
+        // otherwise Next would never run the callback variant of `next/after`
+        res.emit('close')
+
+        // if waitUntil is not available, we have to keep response stream open until background promises are resolved
+        // to ensure that all background work executes
+        await requestContext.backgroundWorkPromise
```

Comment:
This used to be wrapped in `!context.waitUntil`, is that still needed?
If not, could we update the comment? `if waitUntil is not available...`

Reply:
I'll update the comment.

When `context.waitUntil` is available we don't track background work and instead let `context.waitUntil` handle that. So this ends up being no-op (`await Promise.allSettled([])`) when `context.waitUntil` is available

Reply:
Some clarification in https://github.com/opennextjs/opennextjs-netlify/pull/2717/commits/c938bb6aefd1f4f94c6c0d0f748e0b1bddda0239